analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.Classifier.cs

```diff
@@ -1286,4 +1286,26 @@ public void M(object o)
                 }
             }
             """, allowSemanticModel);
+
+    [DataTestMethod]
+    [DataRow("var d = [u:dateTimePointer]->Date;", false)]
```

Comment:
Unknown is the default tokentype and is used for all tokens that are not of any of the other types:
```cs
  public enum TokenType {
    [pbr::OriginalName("UNKNOWN_TOKENTYPE")] UnknownTokentype = 0,
    [pbr::OriginalName("TYPE_NAME")] TypeName = 1,
    [pbr::OriginalName("NUMERIC_LITERAL")] NumericLiteral = 2,
    [pbr::OriginalName("STRING_LITERAL")] StringLiteral = 3,
    [pbr::OriginalName("KEYWORD")] Keyword = 4,
    [pbr::OriginalName("COMMENT")] Comment = 5,
  }
```
"Unknown" tokens are skipped when the info is written to the disc because there is no difference.